### DISCUSSION TITLE
How to disable  SelectFilter or set the value  become 'all', from other SelectFilter ?.

### DISCUSSION THREAD

**Discussion by jigujigu** [filamentphp/filament#9759] [Help]
How to disable  SelectFilter or set the value  become 'all', from other SelectFilter ?.
Let say : I have 2 SelectFilters (type-1 and type-2).

I want set the value in 1st SelectFilter  (type-1) to become 'all' or just disable in.  If Type-2 have select any value (not 'all').

Thanks

```
                SelectFilter::make('type_1')->label('Type-1')
                    ->options(Category::distinct()
                        ->pluck('type_1', 'type_1'))
                    ->query(
                        fn (Builder $query, array $data) => $query->when(
                            $data['value'],
                            fn (Builder $query, $type_1): Builder => $query->whereRelation('category', 'type_1', $type_1)
                        ),
                    ),

                SelectFilter::make('type_2')->label('Type-2')
                    ->options(Category::distinct()
                        ->pluck('type_2', 'type_2'))
                    ->query(
                        fn (Builder $query, array $data) => $query->when(
                            $data['value'],
                            fn (Builder $query, $type_2): Builder => $query->whereRelation('category', 'type_2', $type_2)
                        ),
                    ),
```
Thanks

**Comment by leandrocfe** [comment]
Maybe this?

```php
SelectFilter::make('type_2')->label('Type-2')
    ->options(Category::distinct()
        ->pluck('type_2', 'type_2'))
    ->query(
        fn (Builder $query, array $data, $livewire) => $query->when(
            $data['value'],
            function (Builder $query) use ($data, $livewire): Builder {

                $livewire->removeTableFilter('type_1');

                return $query->whereRelation('category', 'type_2', $data['value']);
            }
        ),
    )
```

**Reply by jigujigu** [reply]
the 1st 'SelectFilter' component is successfully setback to null (all),. but active-filter still active.

see at the line of 'active filters' .
![Recording 2023-11-22 100723](https://github.com/filamentphp/filament/assets/6214964/d2d50db2-77e5-4589-87ba-3a7aee960c46)

And how I can disabled/enabled the 1st SelectFilter component after check or uncheck 2nd filter ?
To make it, more simple. I change the 2nd SelectFilter become Filter (checkbox)

                SelectFilter::make('type_1')->label('Type')
                    ->options(Category::distinct()
                        ->pluck('type_1', 'type_1'))
                    ->query(
                        fn (Builder $query, array $data) => $query->when(
                            $data['value'],
                            fn (Builder $query, $type_1): Builder => $query->whereRelation('category', 'type_1', $type_1)
                        ),
                    ),

                Filter::make('is_overstock')->label('Over-Stock > 500')
                    ->query(function (Builder $query, $livewire): Builder {
                        $livewire->removeTableFilter('type_1');
                        return $query->where('qty', '>=', 500);
                    }),

**Reply by leandrocfe** [reply]
try
```php
$livewire->removeTableFilter('type_1');
$livewire->resetTableColumnSearch('type_1');
```

**Reply by jigujigu** [reply]
Hi leandroce,

I tried, but not working. And filter 'type_1' still active.
```
                Filter::make('is_overstock')->label('Over-Stock > 500')
                    ->query(function (Builder $query, $livewire): Builder {
                        $livewire->removeTableFilter('type_1');
                        $livewire->resetTableColumnSearch('type_1');
                        return $query->where('qty', '>=', 500);
                    }),
```

![image](https://github.com/filamentphp/filament/assets/6214964/88e44596-57cb-4809-914f-8be9683184cc)

**Reply by jigujigu** [reply]
Anyone can help me?
Thanks

**Comment by agencetwogether** [comment]
I know this is old post but did you find a way to achieve this since  @jigujigu ?
